local_vars/local_vars.cc

```diff
@@ -226,35 +232,246 @@ class LocalNameInserter {
     }
 
     ast::ExpressionPtr postTransformSend(core::MutableContext ctx, ast::ExpressionPtr tree) {
+        // Replace ZSuperArgs with an explicit forwarding of the enclosing method's arguments.
+
         auto &original = ast::cast_tree_nonnull<ast::Send>(tree);
-        if (original.args.size() == 1 && ast::isa_tree<ast::ZSuperArgs>(original.args[0])) {
```

Comment:
You _might_ want to pre-emptively move this `super` logic into a helper function. I don't think that we do anything other than super stuff in `postTransformSend` right now, but I imagine the first person who needs to touch this code after your change lands for something not super-related will appreciate having already had things teased apart.